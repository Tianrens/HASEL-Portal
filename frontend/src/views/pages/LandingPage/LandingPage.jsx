import React from 'react';
import { content, QnA, loginContainer } from './LandingPage.module.scss';
import LoginButton from '../../components/buttons/LoginButton';
import HeroPageTemplate from '../../components/templates/HeroPageTemplate/HeroPageTemplate';

function loginCallback() {
    console.log('login callback');
}

function loginFailCallback(error) {
    console.log(`login fail callback: ${error}`);
}

function LandingPage() {
    return (
        <HeroPageTemplate>
            <div className={content}>
                <h1>HASEL Lab</h1>
                <div className={QnA}>
                    <h2>What is HASEL?</h2>
                    <p>
                        The Human Aspects in Software Engineering Lab (HASEL)
                        researches projects related to improving software
                        practices and processes, with a focus on the human
                        aspects of software engineering. We have several high
                        performance machine learning computers available for use
                        in the lab. They can be accessed remotely through SSH as
                        well.
                    </p>
                    <h2>How do I gain access?</h2>
                    <p>
                        Login with your university connected Google account
                        below. Then enter your details and apply for a
                        workstation. After access is granted you will receive an
                        email with your workstation details. You can then use
                        this site to book the lab computers.
                    </p>
                </div>
            </div>
            <div className={loginContainer}>
                <LoginButton
                    callback={loginCallback}
                    errorCallback={loginFailCallback}
                />
            </div>
        </HeroPageTemplate>
    );
}

export default LandingPage;
