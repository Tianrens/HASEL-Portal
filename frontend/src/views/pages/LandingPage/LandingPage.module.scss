@import '../../../assets/colours';

h1 {
    font-size: 3rem;
    color: $heading;
}

h2 {
    font-size: 2rem;
    margin-top: 5rem;
    width: 60%;
}

p {
    font-size: 1.2rem;
}

.content {
    margin: 4rem;
}

.loginContainer {
    display: flex;
    align-items: stretch;
    justify-content: center;
    flex: 1;
    button {
        height: fit-content;
        padding: 1rem 4rem;
    }
}

.QnA {
    color: $body;
    margin-left: 2rem;
    width: 80%;
}
