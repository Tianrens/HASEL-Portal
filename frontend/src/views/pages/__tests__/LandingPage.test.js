import React from 'react';
import renderer from 'react-test-renderer';
import { StylesProvider } from '@material-ui/core';
import { SnackbarProvider } from 'notistack';
import LandingPage from '../LandingPage/LandingPage';

test('Landing Page renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <SnackbarProvider>
                <LandingPage />
            </SnackbarProvider>
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
