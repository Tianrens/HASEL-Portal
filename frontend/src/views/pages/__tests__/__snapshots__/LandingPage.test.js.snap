// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page renders properly 1`] = `
<div
  className="fullPage"
>
  <div
    className="heroPattern"
  >
    <div
      className="container"
    >
      <div
        className="content"
      >
        <h1>
          HASEL Lab
        </h1>
        <div
          className="QnA"
        >
          <h2>
            What is HASEL?
          </h2>
          <p>
            The Human Aspects in Software Engineering Lab (HASEL) researches projects related to improving software practices and processes, with a focus on the human aspects of software engineering. We have several high performance machine learning computers available for use in the lab. They can be accessed remotely through SSH as well.
          </p>
          <h2>
            How do I gain access?
          </h2>
          <p>
            Login with your university connected Google account below. Then enter your details and apply for a workstation. After access is granted you will receive an email with your workstation details. You can then use this site to book the lab computers.
          </p>
        </div>
      </div>
      <div
        className="loginContainer"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained button undefined"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Login
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="rightSide"
  >
    <div
      className="bgImg"
    />
  </div>
</div>
`;
